src/doc/user_data.h

```diff
@@ -10,12 +10,18 @@
 
 #include "doc/color.h"
 
+#include <map>
 #include <string>
+#include <variant>
+#include <vector>
 
 namespace doc {
 
   class UserData {
   public:
+    typedef std::variant<bool, int, std::string> PrimitiveVariant;
+    typedef std::variant<PrimitiveVariant, std::vector<PrimitiveVariant>> Variant;
```

Comment:
This is not possible because std::vector<> needs a complete type/with well-known size at compile time. And generally std:: types weren't created to be derived (but that's other issue). Something like std::vector<VariantStruct*> would be possible.

Anyway I'm not sure about this. Probably to keep this (and the serialization) simple, we should just support one level of properties (without maps/objects).